addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,712 @@
+                }
+            if not taxes and move.l10n_gr_edi_inv_type not in TYPES_WITH_VAT_CATEGORY_8:
+                errors[f'l10n_gr_edi_line_{line_no}_missing_tax'] = {
+                    'message': _('Missing tax on line %s.', line_no),
+                }
+            if len(taxes) == 1 and taxes.amount == 0 and not line.l10n_gr_edi_tax_exemption_category:
+                errors[f'l10n_gr_edi_line_{line_no}_missing_tax_exempt'] = {
+                    'message': _('Missing MyDATA Tax Exemption Category for line %s.', line_no),
+                }
+            if len(taxes) == 1 and taxes.amount not in VALID_TAX_AMOUNTS:
+                errors[f'l10n_gr_edi_line_{line_no}_invalid_tax_amount'] = {
+                    'message': _(source='Invalid tax amount for line %(line_no)s. The valid values are %(valid_values)s.',
+                                 line_no=line_no,
+                                 valid_values=', '.join(str(tax) for tax in VALID_TAX_AMOUNTS)),
+                }
+        return errors
+
+    def _l10n_gr_edi_get_pre_error_string(self):
+        self.ensure_one()
+        pre_error = self._l10n_gr_edi_get_pre_error_dict()
+        error_messages = (error_val['message'] for error_val in pre_error.values())
+        return '\n'.join(error_messages)
+
+    @api.model
+    def _l10n_gr_edi_handle_send_result(self, result, xml_vals):
+        """
+        Handle the result object received from sending xml to myDATA.
+        Create the related error/sent document with the necessary values.
+        """
+        move_xml_map = {}  # Dictionary mapping of ``move_id`` -> ``xml_content``.
+        for invoice_vals in xml_vals['invoice_values_list']:
+            single_xml_vals = {'invoice_values_list': [invoice_vals]}
+            move = invoice_vals['__move__']
+            xml_template = 'l10n_gr_edi.mydata_invoice' if move.is_sale_document(include_receipts=True) else 'l10n_gr_edi.mydata_expense_classification'
+            xml_content = self._l10n_gr_edi_generate_xml_content(xml_template, single_xml_vals)
+            move_xml_map[move] = xml_content
+
+        move_ids = list(move_xml_map.keys())
+
+        if 'error' in result:
+            # If the request failed at this stage, it is probably caused by connection/credentials issues.
+            # In such case, we don't need to attach the xml here as it won't be helpful for the user.
+            for move in move_ids:
+                move._l10n_gr_edi_create_error_document(result['error'])
+        else:
+            for result_id, result_dict in result.items():
+                move = move_ids[result_id]
+                xml_content = move_xml_map[move]
+                document_values = {**result_dict, 'xml_content': xml_content}
+                # Delete previous error documents
+                move.l10n_gr_edi_document_ids.filtered(lambda d: d.state in ('invoice_error', 'bill_error') and not d.attachment_id).unlink()
```

Comment:
```suggestion
                move.l10n_gr_edi_document_ids.filtered(lambda d: d.state in ('invoice_error', 'bill_error')).unlink()
```
Why keeping documents in error? If it has failed, the attachment is linked to the document do we don't care